aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,98 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void pre_check_gradient(const Tensor& self, IntArrayRef dim, int64_t spacing_size, int64_t edge_order, bool spacing_existence, bool dim_existence) {
```

Comment:
lets call this function in signatures instead of helpers, so we don't need send boolen variables here but it can take optional variables.